### PR TITLE
(PC-7200) Get user credit and compare it with offer price

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7200

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/bookOffer/services/utils.ts [hunk 1]
```diff
@@ -27,15 +27,17 @@ export enum OfferStatus {
...
-  stocksDates: { [date: string]: OfferStockResponse[] }
+  stocksDates: { [date: string]: OfferStockResponse[] },
+  userRemainingCredit: number | null
 ): { status: OfferStatus; price: string | null } => {
   const stocksByDate = stocksDates[formatToSlashedFrenchDate(date.toString())]
   if (!stocksByDate) return { status: OfferStatus.NOT_OFFERED, price: null }
 
   const prices = stocksByDate.map((stock) => stock.price)
-  const price = formatToFrenchDecimal(Math.min(...prices)).replace(' ', '')
+  const price = Math.min(...prices)
```
Comment (antoinewg) [inline src/features/bookOffer/services/utils.ts:37]: regarde la fonction `getOfferPrice`

File: src/features/bookOffer/services/utils.ts [hunk 2]
```diff
@@ -27,15 +27,17 @@ export enum OfferStatus {
...
   const offerIsBookable = stocksByDate.some((stock) => stock.isBookable)
-
-  if (offerIsBookable) return { status: OfferStatus.BOOKABLE, price }
-  return { status: OfferStatus.NOT_BOOKABLE, price }
+  const hasEnoughCredit = userRemainingCredit ? userRemainingCredit > price : false
+  if (offerIsBookable && hasEnoughCredit)
+    return { status: OfferStatus.BOOKABLE, price: formatToFrenchDecimal(price).replace(' ', '') }
```
Comment (antoinewg) [inline src/features/bookOffer/services/utils.ts:41]: calcule `const price = formatToFrenchDecimal(getOfferPrice(stocks)).replace(' ', '')` dessus pour éviter de répeter le replace => plus concis => plus lisible.

File: src/features/bookOffer/components/BookingEventChoices.tsx [hunk 3]
```diff
@@ -20,6 +22,20 @@ interface Props {
...
 export const BookingEventChoices: React.FC<Props> = ({ dismissModal, stocks }) => {
   const { bookingState, dispatch } = useBooking()
+  const { data: user } = useUserProfileInfo()
+
+  if (!user) return <React.Fragment />
+  const depositVersion = user.depositVersion && user.depositVersion === 1 ? 1 : 2
+  const expenses = user.expenses
+  const walletBalance = computeWalletBalance(user.expenses)
+
+  const expense = sortExpenses(depositVersion, expenses).find(
```
Comment (antoinewg) [inline src/features/bookOffer/components/BookingEventChoices.tsx:32]: on veut pas uniquement le crédit sortie?

File: src/features/bookOffer/components/BookingEventChoices.tsx [hunk 4]
```diff
@@ -20,6 +22,20 @@ interface Props {
 
 export const BookingEventChoices: React.FC<Props> = ({ dismissModal, stocks }) => {
   const { bookingState, dispatch } = useBooking()
+  const { data: user } = useUserProfileInfo()
+
+  if (!user) return <React.Fragment />
+  const depositVersion = user.depositVersion && user.depositVersion === 1 ? 1 : 2
+  const expenses = user.expenses
+  const walletBalance = computeWalletBalance(user.expenses)
```
Comment (antoinewg) [inline src/features/bookOffer/components/BookingEventChoices.tsx:30]: regarde le hook `useAvailableCredit` => il fait exactement ça, dommage de ne pas le réutiliser. Si tu dois le changer un peu, crée un hook et met le dans le même fichier, ça évite d'en avoir un peu de partout

File: src/features/bookOffer/components/BookingEventChoices.tsx (hunk 3 above)
Comment (emmajmn) [reply src/features/bookOffer/components/BookingEventChoices.tsx:32]: ça sera gérer après le ticket back que victor va faire pour gérer justement correctement tous les crédits

File: src/features/bookOffer/services/utils.ts (hunk 1 above)
Comment (emmajmn) [reply src/features/bookOffer/services/utils.ts:37]: modif ok

File: src/features/bookOffer/services/utils.ts (hunk 2 above)
Comment (emmajmn) [reply src/features/bookOffer/services/utils.ts:41]: yes modif ok

File: src/features/bookOffer/components/BookingEventChoices.tsx (hunk 4 above)
Comment (emmajmn) [reply src/features/bookOffer/components/BookingEventChoices.tsx:30]: ok je fais la modif

**Review by antoinewg — APPROVED** [review]

**Pull request merged**